### PR TITLE
Simplify prepare-commit-msg

### PR DESCRIPTION
**Pull request by camsteffen**
* Move `main` to the top
* Simplify `add_co_authored_by_trailers`
* Avoid `eval`

### CODE REVIEW

File: prepare-commit-msg [hunk 1]
```diff
@@ -15,32 +27,12 @@ add_jira_issue_id_prefix() {
...
-  [ "$2" = 'commit' ] && exit 0
-
-  add_jira_issue_id_prefix "$@"
-  add_co_authored_by_trailers "$@"
+  git mob --list |
+    sed "s/^/Co-authored-by: /" |
+    tee /dev/tty |
+    sed "s/^/--trailer\n/" |
+    tr '\n' '\0' |
+    xargs -0 git interpret-trailers --if-exists addIfDifferent --in-place "$1"
```
Comment (Mubashwer) [inline prepare-commit-msg:35]: Currently a newline is printed after the Co-authored-by trailers are printed. Could you please add the following:

```suggestion
    xargs -0 git interpret-trailers --if-exists addIfDifferent --in-place "$1"
  printf "\n" 
```

image

**Review by Mubashwer — CHANGES_REQUESTED** [review]
@camsteffen Thank you for your contribution and the nice improvements! Its more concise and idiomatic. Requesting a minor change in the script. 

Could you also prefix your commit with `refactor: ` as the repo uses [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)?

File: prepare-commit-msg (hunk 1 above)
Comment (camsteffen) [reply prepare-commit-msg:35]: Welp I went ahead and opened another PR with a new test! Perhaps we can merge that first.

Comment (Mubashwer) [reply prepare-commit-msg:35]: The integration test is now failing: `stderr="tee: /dev/tty: No such device or address\n"`

Comment (camsteffen) [reply prepare-commit-msg]: Okay I think it's ready now. I stopped using /dev/tty, that seems kinda iffy.

Thanks so much for reviewing and accepting changes!

**Review by Mubashwer — APPROVED** [review]

**Pull request merged**